src/Compilers/CSharp/Portable/Syntax/NullableContextStateMap.cs

```diff
@@ -133,7 +133,7 @@ private static ImmutableArray<NullableContextState> GetContexts(SyntaxTree tree,
                     continue;
                 }
 
-                var position = nn.Location.SourceSpan.End;
+                var position = nn.EndPosition;
```

Comment:
>EndPosition [](start = 34, length = 11)

Shouldn't these be the same?  How could they be different?

Reply:
It seems like syntax nodes created using ReplaceNode have no location, but do have a position.

What I'm still trying to figure out is why the bootstrap build ends up running GetContexts() on a non-CompilationUnit-rooted tree created indirectly from a call to ReplaceNode, but my test only runs GetContexts on the original tree.

Reply:
It appears to be by-design. See the comment on [SyntaxNode.Location](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/Syntax/SyntaxNode.cs#L555). `SyntaxNode.GetLocation()` on the other hand seems to return a proper Location, even on a node created by ReplaceNode.